airflow/providers/amazon/aws/hooks/emr_containers.py

```diff
@@ -0,0 +1,217 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from time import sleep
+from typing import Any, Dict, Optional
+
+from botocore.exceptions import ClientError
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class EMRContainerHook(AwsBaseHook):
+    """
+    Interact with AWS EMR Virtual Cluster to run, poll jobs and return job status
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+
+    :param virtual_cluster_id: Cluster ID of the EMR on EKS virtual cluster
+    :type virtual_cluster_id: str
+    """
+
+    INTERMEDIATE_STATES = (
+        "PENDING",
+        "SUBMITTED",
+        "RUNNING",
+    )
+    FAILURE_STATES = (
+        "FAILED",
+        "CANCELLED",
+        "CANCEL_PENDING",
+    )
+    SUCCESS_STATES = ("COMPLETED",)
+
+    def __init__(self, *args: Any, virtual_cluster_id: Optional[str] = None, **kwargs: Any) -> None:
+        super().__init__(client_type="emr-containers", *args, **kwargs)  # type: ignore
+        self.virtual_cluster_id = self._get_virtual_cluster_id(virtual_cluster_id, self.aws_conn_id)
+
+    def _get_virtual_cluster_id(self, virtual_cluster_id: str, aws_conn_id: str):
+        if virtual_cluster_id is not None:
+            return virtual_cluster_id
+
+        if aws_conn_id is not None:
+            conn = self.get_connection(aws_conn_id)
+            cluster_id = conn.extra_dejson.get('virtual_cluster_id')
```

Comment:
This is not standard behavior for AWS Base Hook, so you need to write a guide on how to configure this connection. Similarly, we have [a GCP connection](http://airflow.apache.org/docs/apache-airflow-providers-google/stable/connections/gcp.html) and [a GCP SSH connection](http://airflow.apache.org/docs/apache-airflow-providers-google/stable/connections/gcp_ssh.html).

Reply:
OK, will do - wondering if I should remove if it's not standard. I saw it used in another connector and thought it could be useful to be able to configure the virual_cluster_id and role_arn via a connection.

Reply:
I have no opinion on this matter. You can delete it if you want because then the user can use defaults_args to limit duplications.

Reply:
OK, going to remove it since it's non-standard.